crates/pathfinder/src/state/class_hash.rs

```diff
@@ -117,39 +117,75 @@ fn compute_class_hash0(mut contract_definition: json::ContractDefinition<'_>) ->
             Ok(())
         })?;
 
-    /// Insert's a space ": " -> " : " for every tuple value in `cairo_type`. This is required by StarkNet 0.10 for
-    /// some obscure backwards compatibility reason for older contracts.
-    fn insert_space(object: &mut serde_json::Map<String, serde_json::Value>) {
+    fn add_extra_space_to_cairo_named_tuples(value: &mut serde_json::Value) {
+        match value {
+            serde_json::Value::Array(v) => walk_array(v),
+            serde_json::Value::Object(m) => walk_map(m),
+            _ => {}
+        }
+    }
+
+    fn walk_array(array: &mut [serde_json::Value]) {
+        for v in array.iter_mut() {
+            add_extra_space_to_cairo_named_tuples(v);
+        }
+    }
+
+    fn walk_map(object: &mut serde_json::Map<String, serde_json::Value>) {
         for (k, v) in object.iter_mut() {
-            if k == "cairo_type" {
-                // Check that the value is a tuple.
-                if let Some(tuple_inner) = v
-                    .as_str()
-                    .and_then(|v| v.strip_prefix('(').and_then(|v| v.strip_suffix(')')))
+            match v {
+                serde_json::Value::String(s) => {
+                    let new_value = add_extra_space_to_named_tuple_type_definition(k, s);
+                    if new_value.as_ref() != s {
```

Comment:
I think this could be `if new_value.is_owned()` ... Since we are adding characters I can't see how this could be done with an in-place modification.

Reply:
Maybe it isn't so easy.